### ISSUE TITLE
v8 - "browser cookie" error

### ISSUE THREAD

**Issue by armchairancap** [Tribler/tribler#8227]
I start Tribler 8.0.1 from the CLI (python3 src/run_tribler.py) and it opens UI in my default browser. Good.

Now I use the same `http://localhost` link to UI from another browser on the same host where Tribler is running, and I get this error about missing a cookie.

![image](https://github.com/user-attachments/assets/f400032c-a262-4286-9b51-d5180ecd199d)

What cookie? 
How do I get it? 
How do I set it in another browser? 
Is that documented somewhere?

I don't want to change my default system browser just because of this problem, so I'd be able to provide this in the UI link or somehow get it from the Tribler tray (copy cookie to clipboard?) or use some other way.

**Comment by armchairancap** [comment]
Answer: should be set in headers.

**Comment by qstokkink** [comment]
@armchairancap We even have three options, you can choose whichever is most convenient for you:

https://github.com/Tribler/tribler/blob/faef65d183d0f7c69c2592f07f2799d2c4142203/src/tribler/core/restapi/rest_manager.py#L96

You already found the `X-Api-Key` header but you can also pass the key in the URL (using `.../endpoint?key=`) or put the key in the `api_key` cookie.

For single requests, I usually prefer passing the key in the URL.

**Comment by HuckleHubbing** [comment]
Hi, I'm having this "Failed to connect to Tribler" cookie error when I try to open this (127.0.0.1:<port>) in a different browser.  But I don't understand the answer provided here. Any chance someone can dumb it down for me a bit?

**Comment by armchairancap** [comment]
Try to append `?key=YOURKEY` to the link where YOURKEY may be the default `changeme` or whatever you set it to.
I describe this in [here](https://github.com/armchairancap/xx-tribler-haven/?tab=readme-ov-file#xx-network-haven---tribler-proof-of-concept) (use CTRL+F to find `changeme`).
